dbt/adapters/bigquery/connections.py

```diff
@@ -309,6 +309,8 @@ def get_google_credentials(cls, profile_credentials) -> GoogleCredentials:
 
         elif method == BigQueryConnectionMethod.SERVICE_ACCOUNT_JSON:
             details = profile_credentials.keyfile_json
+            private_key = details.get("private_key", "").replace("\\n", "\n")
```

Comment:
I feel like we should do this when `profile_credentials` is created so that we don't run into a scenario where `keyfile_json` looks like one thing one place, but another someplace else. If we're saying these two strings are the same (I agree), then we should cleanse it on the way in, not on the way out.

Reply:
Thanks for giving this a look @mikealfare !

Where are you thinking you'd like to see this logic moved to?

Reply:
What is `profile_credentials`? Is that a `BigQueryCredentials` instance where `keyfile_json` is this: https://github.com/dbt-labs/dbt-bigquery/blob/3761b408e8850a306dca0ba874483d0756486deb/dbt/adapters/bigquery/connections.py#L124? If so, I'd probably create a `__post_init__()` that makes that update if `keyfile_json` is populated. But there's a lot of hand waving there. You could always try it and see if it works.

Reply:
`__post_init__` works! Thank you for the suggestion.